media-proxy/src/mesh/st2110_20rx.cc

```diff
@@ -0,0 +1,65 @@
+#include "st2110rx.h"
+
+namespace mesh::connection {
+
+st_frame *ST2110_20Rx::get_frame(st20p_rx_handle h)
+{
+    return st20p_rx_get_frame(h);
+};
+
+int ST2110_20Rx::put_frame(st20p_rx_handle h, st_frame *f)
+{
+    return st20p_rx_put_frame(h, f);
+};
+
+st20p_rx_handle ST2110_20Rx::create_session(mtl_handle h, st20p_rx_ops *o)
+{
+    return st20p_rx_create(h, o);
+};
+
+int ST2110_20Rx::close_session(st20p_rx_handle h)
+{
+    return st20p_rx_free(h);
+};
+
+Result ST2110_20Rx::configure(context::Context &ctx, const std::string &dev_port,
+                              const MeshConfig_ST2110 &cfg_st2110,
+                              const MeshConfig_Video &cfg_video)
+{
+    if (cfg_st2110.transport != MESH_CONN_TRANSPORT_ST2110_20) {
+        set_state(ctx, State::not_configured);
+        return set_result(Result::error_bad_argument);
+    }
+
+    if (configure_common(ctx, dev_port, cfg_st2110))
+        return set_result(Result::error_bad_argument);
+
+    ops.port.payload_type = ST_APP_PAYLOAD_TYPE_ST20;
+    ops.width = cfg_video.width;
+    ops.height = cfg_video.height;
+    ops.fps = st_frame_rate_to_st_fps(cfg_video.fps);
+    ops.transport_fmt = ST20_FMT_YUV_422_PLANAR10LE;
+
+    if (mesh_video_format_to_st_format(cfg_video.pixel_format, ops.output_fmt))
+        return set_result(Result::error_bad_argument);
+
+    ops.device = ST_PLUGIN_DEVICE_AUTO;
+
+    transfer_size = st_frame_size(ops.output_fmt, ops.width, ops.height, false);
+    if (transfer_size == 0)
+        return set_result(Result::error_bad_argument);
```

Comment:
State will stay `configuring` here.

Reply:
Fixed in https://github.com/OpenVisualCloud/Media-Communications-Mesh/pull/261/commits/4db69d1907a638e19df895bab4b5914102226a85